pkg/prometheus/agent/statefulset.go

```diff
@@ -317,6 +317,8 @@ func makeStatefulSetSpec(
 				TopologySpreadConstraints:     prompkg.MakeK8sTopologySpreadConstraint(finalSelectorLabels, cpf.TopologySpreadConstraints),
 				HostAliases:                   operator.MakeHostAliases(cpf.HostAliases),
 				HostNetwork:                   cpf.HostNetwork,
+				DNSPolicy:                     k8sutil.ConvertDNSPolicy(cpf.DNSPolicy),
```

Comment:
We should rather only set the fields when they're not nil to avoid undesired side-effects (the same for all controllers).

```golang
    if cpf.DNSPolicy != nil {
        spec.DNSPolicy = appsv1.DNSPolicy(*cpf.DNSPolicy)
    }

    if cpf.DNSConfig != nil {
        spec.DNSConfig = k8sutil.ConvertToK8sDNSConfig(*cpf.DNSConfig)
    }
```

Reply:
@simonpasquier I don't fully follow where I need to change this. Could you elaborate please?